### PR TITLE
[feat]: [PIE-3765]: Add API to get Input Set by ID regardless of type

### PR DESCRIPTION
**Pull request by namanvermah**
Right now, we have separate GET calls for getting an input set and getting an overlay input set. However, if someone only has an input set ID, they won't have any way to decide which API to call. To solve this, I have added an API that takes an identifier, and based upon the type of the input set, the response will have the corresponding field filled.

Why this was needed now: triggers can now have an input set reference, they'll need this API.

This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/33234)

### CODE REVIEW

**Review by naidu-harness — APPROVED** [review]
:lgtm:

> Reviewed 5 of 5 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33234#-:-N2Fa-oq8dc2qdVF_HEE:bnfp4nl)* status: 5 file(s) to review. 1 of 2 approvals obtained.

**Review by archit-harness — CHANGES_REQUESTED** [review]
> Reviewed 4 of 5 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33234)* status: 1 file(s) to review. 1 change requests. 1 of 2 approvals obtained. 

___
*[`800-pipeline-service/src/main/java/io/harness/pms/ngpipeline/inputset/beans/resource/InputSetByIDResponseDTO.java` line 23 at r1](https://reviewable.io/reviews/harness/harness-core/33234#-N2Fd-_ZA5EzYZ-UGiJY:-N2Fd-_ZA5EzYZ-UGiJZ:b-6w6lrm) ([raw file](https://github.com/harness/harness-core/blob/1cbc892c4edafa659adeec366ad5da6df7dbc3c1/800-pipeline-service/src/main/java/io/harness/pms/ngpipeline/inputset/beans/resource/InputSetByIDResponseDTO.java#L23)):*
> ```Java
> @JsonInclude(JsonInclude.Include.NON_NULL)
> @ApiModel("InputSetByIDResponse")
> public class InputSetByIDResponseDTO {
> ```

@Schema missing and on fields as well and its inner fields with description, reason its a customer endpoint else mark the API as hidden if not customer usable



_Code quote:_
```Java
nputSetByIDResponseDTO
```

**Review by namanvermah — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33234)* status: 1 file(s) to review. 1 change requests. 1. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`800-pipeline-service/src/main/java/io/harness/pms/ngpipeline/inputset/beans/resource/InputSetByIDResponseDTO.java` line 23 at r1](https://reviewable.io/reviews/harness/harness-core/33234#-N2Fd-_ZA5EzYZ-UGiJY:-N2Fd_dm-FUXBXK5ZyHS:by5cbpj) ([raw file](https://github.com/harness/harness-core/blob/1cbc892c4edafa659adeec366ad5da6df7dbc3c1/800-pipeline-service/src/main/java/io/harness/pms/ngpipeline/inputset/beans/resource/InputSetByIDResponseDTO.java#L23)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

@Schema missing and on fields as well and its inner fields with description, reason its a customer endpoint else mark the API as hidden if not customer usable
</blockquote></details>

The API is marked as hidden yes

**Review by archit-harness — APPROVED** [review]
> Reviewed 1 of 5 files at r1.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33234)* status: :shipit: complete! ✓

___
*[`800-pipeline-service/src/main/java/io/harness/pms/ngpipeline/inputset/beans/resource/InputSetByIDResponseDTO.java` line 23 at r1](https://reviewable.io/reviews/harness/harness-core/33234#-N2Fd-_ZA5EzYZ-UGiJY:-N2Fe-fmEcRCtmBQbjPx:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/1cbc892c4edafa659adeec366ad5da6df7dbc3c1/800-pipeline-service/src/main/java/io/harness/pms/ngpipeline/inputset/beans/resource/InputSetByIDResponseDTO.java#L23)):*
<details><summary><i>Previously, namanvermah (Naman Verma) wrote…</i></summary><blockquote>

The API is marked as hidden yes
</blockquote></details>

done

File: 800-pipeline-service/src/main/java/io/harness/pms/ngpipeline/inputset/resources/InputSetResourcePMS.java [hunk 1]
```diff
@@ -135,6 +136,48 @@ public class InputSetResourcePMS {
...
+      @NotNull @QueryParam(NGCommonEntityConstants.ORG_KEY) @OrgIdentifier @Parameter(
+          description = PipelineResourceConstants.ORG_PARAM_MESSAGE) String orgIdentifier,
+      @NotNull @QueryParam(NGCommonEntityConstants.PROJECT_KEY) @ProjectIdentifier @Parameter(
+          description = PipelineResourceConstants.PROJECT_PARAM_MESSAGE) String projectIdentifier,
+      @Parameter(description = InputSetSchemaConstants.PIPELINE_ID_FOR_INPUT_SET_PARAM_MESSAGE) @NotNull @QueryParam(
+          NGCommonEntityConstants.PIPELINE_KEY) @ResourceIdentifier String pipelineIdentifier,
+      @BeanParam GitEntityFindInfoDTO gitEntityBasicInfo) {
+    log.info(String.format("Retrieving input set with identifier %s for pipeline %s in project %s, org %s, account %s",
+        inputSetIdentifier, pipelineIdentifier, projectIdentifier, orgIdentifier, accountId));
+    Optional<InputSetEntity> inputSetEntity = pmsInputSetService.get(
```
Comment (harshjain12) [inline 800-pipeline-service/src/main/java/io/harness/pms/ngpipeline/inputset/resources/InputSetResourcePMS.java:166]: can we make it list

File: 800-pipeline-service/src/main/java/io/harness/pms/ngpipeline/inputset/beans/resource/InputSetByIDResponseDTO.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import lombok.experimental.FieldDefaults;
+
+@OwnedBy(PIPELINE)
+@Value
+@Builder
+@FieldDefaults(level = AccessLevel.PRIVATE)
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@ApiModel("InputSetByIDResponse")
+public class InputSetByIDResponseDTO {
```
Comment (harshjain12) [inline 800-pipeline-service/src/main/java/io/harness/pms/ngpipeline/inputset/beans/resource/InputSetByIDResponseDTO.java:23]: move it to common module

**Comment by harshjain12** [comment]
validation code can also be moved to common module

**Comment by namanvermah** [comment]
Closing this because this API is not what triggers actually need. The merge call can suffice all usecases

**Pull request closed without merging**